aladdin/lib/cache.py

```diff
@@ -38,7 +39,7 @@ def certificate_cache(func):
     @functools.wraps(func)
     @clear_on_error()
     def wrapper(certificate_scope):
-        cache = shelve.open(cache_path)
+        cache = shelve.open(str(cache_path))
```

Comment:
seems like some python versions expect this explicitly as a string instead of a path